core/src/main/scala/org/apache/spark/resource/ResourceInformation.scala

```diff
@@ -35,3 +41,25 @@ class ResourceInformation(
 
   override def toString: String = s"[name: ${name}, addresses: ${addresses.mkString(",")}]"
 }
+
+private[spark] object ResourceInformation {
+  /**
+   * Parses a JSON string into a [[ResourceInformation]] instance.
+   */
+  def parseJson(json: String): ResourceInformation = {
+    implicit val formats = DefaultFormats
+    try {
+      parse(json).extract[ResourceInformationJson].toResourceInformation
```

Comment:
Shall we check duplicate addresses for user ?

Reply:
We can add that check later. It is beyond the scope of this refactor.